### PR TITLE
forget: group-by-tags-only

### PR DESCRIPTION
**Pull request by mungomat**
This is not complete - what if "restic forget --group-by-tags --group-by-tags-only" is given?

### CODE REVIEW

**Comment by fawick** [comment]
Travis fails because the manpage is not up-to-date in the PR. In your branch, could you please call `restic manpage --output-dir doc/man` and add this to the PR? Thanks!

**Comment by fd0** [comment]
And it needs a rebase, just like the other PR. Let us know if you need help with the rebase.

**Comment by middelink** [comment]
Can I suggest to resolve this in a different way? Instead of adding --group-by-tags-only and possibly --group-by-path-ony or --group-by-host-only, make a flag specifying what your ordering needs to be --group-by=[host,path,tags]. Default to "host,path,tags" to keep functionality the same.

**Comment by fd0** [comment]
@middelink that is an excellent idea, thank you very much! We can just fill the JSON key according to what's present in `--group-by`.

@mungomat Do you think you can tackle this? Or shall I do that?

**Comment by fd0** [comment]
The CI test fail because `cmd_forget.go` is not properly formatted. May I suggest installing the script
`fmt-check` from https://github.com/edsrzf/gofmt-git-hook locally as a
pre-commit hook? It checks formatting before committing automatically, just copy
this script to `.git/hooks/pre-commit`.

File: cmd/restic/cmd_forget.go [hunk 1]
```diff
@@ -91,6 +92,14 @@ func runForget(opts ForgetOptions, gopts GlobalOptions, args []string) error {
 	}
 	snapshotGroups := make(map[string]restic.Snapshots)
 
+	var GroupByTag      bool
+	var GroupByHost     bool
+	var GroupByPath     bool
+
+	GroupByTag = strings.Contains( opts.GroupBy, "tag" )
```
Comment (fd0) [inline cmd/restic/cmd_forget.go]: That is very vague. I'd go for splitting `opts.GroupBy` by `,`, then iterate over the results, and use a `switch` statement that reacts on the strings `tag`, `host` and `path` and returns an error if the string is unknown.

What do you think?

**Review by fd0 — CHANGES_REQUESTED** [review]

**Review by fd0 — APPROVED** [review]

**Comment by fd0** [comment]
Good work, thanks!

**Pull request merged**